src/content/deployment/helm/_index.en.md

```diff
@@ -4,4 +4,106 @@ date: 2020-02-19T21:25:43+01:00
 weight: 20
 ---
 
-This is a placeholder, in the meanwhile you can check out <https://github.com/submariner-io/submariner-charts>
+## Deploying with Helm
+
+### Installing Helm
+
+```bash
+helm repo add submariner-latest https://submariner-io.github.io/submariner-charts/charts
+kubectl -n kube-system create serviceaccount tiller
+kubectl create clusterrolebinding tiller --clusterrole=cluster-admin --serviceaccount=kube-system:tiller
+helm init --service-account tiller
+kubectl -n kube-system rollout status deploy/tiller-deploy
+```
+
+### Setup environment variables
+
+```bash
+export BROKER_NS=submariner-k8s-broker
+export SUBMARINER_PSK=$(LC_CTYPE=C tr -dc 'a-zA-Z0-9' < /dev/urandom | fold -w 64 | head -n 1)
+export SUBMARINER_NS=submariner
+```
+
+If your clusters have overlapping Pod/Service CIDRs, please also set:
+
+```bash
+export GLOBALNET=true
+```
+
+### Deploying the Broker
+
+```bash
+helm install submariner-latest/submariner-k8s-broker \
+             --name "${BROKER_NS}" \
+             --namespace "${BROKER_NS}" \
+             --set submariner.serviceDiscovery=true
+```
+
+Setup additional environment variables
+
+```bash
+export SUBMARINER_BROKER_URL=$(kubectl -n default get endpoints kubernetes \
+    -o jsonpath="{.subsets[0].addresses[0].ip}:{.subsets[0].ports[?(@.name=='https')].port}")
+export SUBMARINER_BROKER_CA=$(kubectl -n "${BROKER_NS}" get secrets \
+    -o jsonpath="{.items[?(@.metadata.annotations['kubernetes\.io/service-account\.name']=='${BROKER_NS}-client')].data['ca\.crt']}")
+export SUBMARINER_BROKER_TOKEN=$(kubectl -n "${BROKER_NS}" get secrets \
+    -o jsonpath="{.items[?(@.metadata.annotations['kubernetes\.io/service-account\.name']=='${BROKER_NS}-client')].data.token}" \
+       | base64 --decode)
+```
+
+### Joining a cluster
+
+```bash
+export KUBECONFIG=kubeconfig-of-the-cluster-to-join
+```
+
+```bash
+export CLUSTER_ID=the-id-of-the-cluster
+export CLUSTER_CIDR=x.x.x.x/x   # the cluster's pod IP CIDR
+export SERVICE_CIDR=x.x.x.x/x   # the cluster's service IP CIDR
+```
+
+If you are using overlapping CIDRs, `export GLOBAL_CIDR=169.254.x.x/x` using an individual non-overlapping
+range for each cluster you join.
+
+```bash
+helm install submariner-latest/submariner \
+        --name submariner \
+        --namespace "${SUBMARINER_NAMESPACE}" \
+        --set ipsec.psk="${SUBMARINER_PSK}" \
+        --set broker.server="${SUBMARINER_BROKER_URL}" \
+        --set broker.token="${SUBMARINER_BROKER_TOKEN}" \
+        --set broker.namespace="${BROKER_NS}" \
```

Comment:
nit: To keep it in sync, shall we use BROKER_NAMESPACE instead of BROKER_NS?

Reply:
BROKER_NS since longer names will trip the linters in some lines.